File: ios/Mobius-Common/Model/MarketOrder.swift
import Foundation

public struct MarketOrder: Decodable {
    public let orderID: String
    public let deliveryDate: String
    public let items: [Item]
    public let totalPrice: Double
    public let currency: String
    public let currentStatus: String

    enum CodingKeys: String, CodingKey {
        case orderID = "order_id"
        case deliveryDate = "delivery_date"
        case items
        case totalPrice = "total_price"
        case currency
        case currentStatus = "current_status"
    }
}

File: Mobius-Demo-Flutter/lib/feature/order/data/model/market_order.dart
import 'package:meta/meta.dart';

import '../../domain/entity/entity.dart';
import 'item.dart';

@immutable
class MarketOrderModel extends MarketOrder {
  const MarketOrderModel({
    required super.orderID,
    required super.deliveryDate,
    required super.items,
    required super.totalPrice,
    required super.currency,
    required super.currentStatus,
  });

  factory MarketOrderModel.fromMap(Map<String, Object?> map) {
    return MarketOrderModel(
      orderID: map['order_id'] as String,
      deliveryDate: map['delivery_date'] as String,
      items: (map['items'] as List<Object?>)
          .map((value) => ItemModel.fromMap(value as Map<String, Object?>))
          .toList(),
      totalPrice: map['total_price'] as double,
      currency: map['currency'] as String,
      currentStatus: map['current_status'] as String,
    );
  }

  Map<String, Object?> toMap() {
    return {
      'order_id': orderID,
      'delivery_date': deliveryDate,
      'items': items,
      'total_price': totalPrice,
      'currency': currency,
      'current_status': currentStatus,
    };
  }
}
